crash in nls

The following input crashes nls, with a panic in the pretty-printer.

```nickel
{
  multipleOf
    : Number -> Dyn -> [| 'Ok, 'Error { message | String, notes] } |]
    = 1
}
```

There's a syntax error, but that isn't the point: it shouldn't cause a crash.

Sure, here's the backtrace. It appears to finish parsing, and then crash while trying to report the parse error. I haven't looked at the AST yet, but I can do that tomorrow.

```
2025-02-05T21:48:14.182 helix_lsp::transport [ERROR] local-nls err <- "stack backtrace:\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   0: rust_begin_unwind\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   1: core::panicking::panic_fmt\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   2: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::next\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   3: nickel_lang_core::pretty::Allocator::record::{{closure}}\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   4: nickel_lang_core::pretty::Allocator::record\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   5: nickel_lang_core::pretty::<impl pretty::Pretty<nickel_lang_core::pretty::Allocator> for &nickel_lang_core::typ::EnumRowF<alloc::boxed::Box<nickel_lang_core::typ::Type>>>::pretty\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   6: pretty::DocBuilder<D,A>::append\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   7: pretty::DocBuilder<D,A>::append\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   8: nickel_lang_core::pretty::<impl pretty::Pretty<nickel_lang_core::pretty::Allocator> for &nickel_lang_core::typ::Type>::pretty\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "   9: nickel_lang_core::pretty::<impl pretty::Pretty<nickel_lang_core::pretty::Allocator> for &nickel_lang_core::typ::Type>::pretty\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  10: nickel_lang_core::pretty::<impl pretty::Pretty<nickel_lang_core::pretty::Allocator> for &nickel_lang_core::typ::Type>::pretty\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  11: nickel_lang_core::pretty::fmt_pretty\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  12: core::fmt::write\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  13: alloc::fmt::format::format_inner\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  14: <nickel_lang_core::error::TypecheckError as nickel_lang_core::error::IntoDiagnostics>::into_diagnostics\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  15: <core::iter::adapters::flatten::FlatMap<I,U,F> as core::iter::traits::iterator::Iterator>::next\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  16: nls::world::World::typecheck\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  17: nls::world::World::parse_and_typecheck\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  18: nls::server::Server::handle_message\n"
2025-02-05T21:48:14.183 helix_lsp::transport [ERROR] local-nls err <- "  19: nls::run\n"
```

--- a/core/src/pretty.rs
+++ b/core/src/pretty.rs
@@ -522,6 +522,7 @@ impl Allocator {
         match id_expr.as_ref() {
             // Nickel will not parse a multiline string literal in this position
             Term::StrChunks(chunks) => self.chunks(chunks, StringRenderStyle::ForceMonoline),
+            Term::ParseError(_) => docs![self, "<parse error>"],
             _ => unimplemented!("Dynamic record fields must be StrChunks currently"),
         }
         .append(self.field_body(field))

--- a/lsp/nls/tests/inputs/no-crash-on-pretty-print.ncl
+++ b/lsp/nls/tests/inputs/no-crash-on-pretty-print.ncl
@@ -1,0 +1,7 @@
+### /main.ncl
+{
+  multipleOf
+    : Number -> Dyn -> [| 'Ok, 'Error { message | String, notes] } |]
+    = 1
+}
+### diagnostic = ["file:///main.ncl"]

--- a/lsp/nls/tests/inputs/no-crash-on-pretty-print.ncl
+++ b/lsp/nls/tests/inputs/no-crash-on-pretty-print.ncl
@@ -1,6 +1,6 @@
 ### /main.ncl
 {
-  multipleOf
+  multiple_of
     : Number -> Dyn -> [| 'Ok, 'Error { message | String, notes] } |]
     = 1
 }

